SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.cpp

```diff
@@ -0,0 +1,238 @@
+			FinishStep();
+		});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Different Server - Verify components "
+				 "on Tick"),
+			FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				CheckComponentsCrossServer(DynamicReplicatedActor, 1, 2);
+			},
+			5.0f);
+
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(2));
+	}
+}
+
+void ASpatialComponentTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialComponentTest, DynamicReplicatedActor);
+}
+
+// Checks the number of components on the servers and clients when an actor does not migrate
+void ASpatialComponentTest::CheckComponents(ASpatialComponentTestActor* Actor, int ExpectedServerId, int ExpectedClient1ComponentCount,
+											int ExpectedClient2ComponentCount)
+{
+	const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+	if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
+	{
+		// Allow it to continue working in Native / Single worker setups.
+		if (GetNumberOfServerWorkers() > 1)
+		{
+			if (LocalWorkerDefinition.Id == ExpectedServerId)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 2), "Server auth - OnAuthorityGained component and OnActorReady component");
+				FinishStep();
+			}
+			else if (Actor->bNetStartup)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 1),
+							"Non-auth servers - Level actors receive OnActorReady only OnAuthorityGained");
+				FinishStep();
+			}
+			else
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 0), "Non-auth servers - Dynamic actors do not receive OnActorReady");
+				FinishStep();
+			}
+		}
+		else // Support for Native / Single Worker.
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, 2),
+						"Native / Single Worker - OnActorReady component and OnAuthorityGained component");
+			FinishStep();
+		}
+	}
+	else // Clients
+	{
+		if (LocalWorkerDefinition.Id == 1)
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, ExpectedClient1ComponentCount), "Client 1");
+			FinishStep();
+		}
+		else if (LocalWorkerDefinition.Id == 2)
+		{
+			RequireTrue(VerifyTestActorComponents(Actor, ExpectedClient2ComponentCount), "Client 2");
+			FinishStep();
+		}
+	}
+}
+
+// Checks the number of components on the servers and clients when an actor migrates
+void ASpatialComponentTest::CheckComponentsCrossServer(ASpatialComponentTestActor* Actor, int StartServerId, int EndServerId)
+{
+	const FWorkerDefinition& LocalWorkerDefinition = GetLocalFlowController()->WorkerDefinition;
+	if (LocalWorkerDefinition.Type == ESpatialFunctionalTestWorkerType::Server)
+	{
+		// Allow it to continue working in Native / Single worker setups.
+		if (GetNumberOfServerWorkers() > 1)
+		{
+			if (LocalWorkerDefinition.Id == StartServerId)
+			{
+				RequireTrue(VerifyTestActorComponents(Actor, 3),
```

Comment:
nit - here and few places in this file - all string literals that get cast to FStrings should be wrapped in the TEXT() macro.

Reply:
Sure, updated the FStrings to be wrapped in TEXT() macro.